### DISCUSSION TITLE
Lifecycle hooks not working in Edit function, Filament resource

### DISCUSSION THREAD

**Discussion by Iainreynard** [filamentphp/filament#12113] [Help] [answered]
### Package

Actions

### Package Version

v3

### How can we help you?

I have a resource in a php Filament App called LoadResource. The standard out of the box Edit action on this resource is working fine - it however does not respond to any lifecycle hooks that I place in the Edit action statement.

My Edit Action is listed below:

`->actions([
                Tables\Actions\ActionGroup::make([
                    Tables\Actions\ViewAction::make(),
                    EditAction::make()
                        -> beforeFormFilled(function () {
                            dd('test');
                        }),
                    Tables\Actions\DeleteAction::make(),
                ])
            ])`

I am using the following:

`use Filament\Tables\Actions\EditAction;
`

I however never get to the function within the beforeFormFilled() closure. Any ideas to assist would be greatly appreciated.

Was expecting the function in the beforeFormFilled() to run.

I tried to place the lifecycle hook on the Edit page of the Resource - that worked. However, I dont know how to reference the model from the Edit page $this-> references the Edit page, not the form or the model. When I try to pass the model to the Edit page, I get an error.

For reference, the following does work - with the hassles mentioned above:

`protected function beforeSave (array $data): void
    {

        dd($data);
    }`

**Comment by leandrocfe** [comment]
You should use this section: https://filamentphp.com/docs/3.x/panels/resources/editing-records

**Reply by Iainreynard** [reply]
Many thanks Leandro. 

I got the functionality working on the Edit page of the Resource using `protected function handleRecordUpdate(Model $record, array $data): Model
    {`

I am curious though why the functionality does not work on the Resource page itself. I was hoping to use something like (as per the documentation): 

`EditAction::make()
    ->using(function (Model $record, array $data): Model {
        
//Some logic
$record->update($data);
 
        return $record;
    })`

but sadly it does not call the function on the main Resource page at all. Is there any reason for this.

Many thanks for your help so far.

**Reply by leandrocfe** [reply, marked as answer]
This will work if you're editing records in a modal action, in a simple resource

https://filamentphp.com/docs/3.x/panels/resources/getting-started/#simple-modal-resources

**Reply by Iainreynard** [reply]
Understood - this is great clarity and has helped a lot - thank you for your assistance. Much appreciated.
